packages/studioCMS/src/coreIntegration.ts

```diff
@@ -0,0 +1,115 @@
+import { checkAstroConfig, studioLogger, studioLoggerOptsResolver, makeFrontend, addIntegrationArray, addIntegrationArrayWithCheck } from './utils';
+import { studioCMSRobotsTXT, studioCMSImageHandler, studioCMSDashboard } from './integrations';
+import { addDts, addVirtualImports, createResolver, defineIntegration } from 'astro-integration-kit';
+import 'astro-integration-kit/types/db';
+import { oResolver, vResolver } from './resolvers';
+import inoxsitemap from '@inox-tools/sitemap-ext';
+import { studioCMSPluginList } from './plugintools';
+import { getStudioConfigFileUrl } from './studiocms-config';
+import { optionsSchema } from './schemas';
+import { version } from '../package.json';
+import { CoreStrings, robotsTXTPreset } from './strings';
+
+// Main Integration
+export default defineIntegration({
+	name: '@astrolicious/studiocms',
+	optionsSchema, 
+	setup({ name, options }) {
+
+		// Register StudioCMS Core as First Plugin
+		studioCMSPluginList.set('@astrolicious/studiocms', { name: '@astrolicious/studiocms', label: 'StudioCMS' });
+		
+		// Create Resolver for Virtual Imports
+		const { resolve } = createResolver(import.meta.url);
+
+		return {
+			hooks: {
+				'astro:db:setup': ({ extendDb }) => { 
+					// Configure `@astrojs/db` integration to include the StudioCMS Database Table Schema
+					extendDb({ configEntrypoint: resolve('./db/config.ts') });
+				},
+				'astro:config:setup': async ( params ) => {
+
+					// Destructure Params
+					const { config: astroConfig, addWatchFile } = params;
+
+					// Watch the StudioCMS Config File for changes (including creation/deletion)
+					addWatchFile(getStudioConfigFileUrl(astroConfig.root));
+
+					// Resolve Options
+					const resolvedOptions = await oResolver(params, options);
```

Comment:
pls can `oResolver` have a better name?

Reply:
any suggestions?

Reply:
`optionResolver` ?

Reply:
teamwork makes the dreamwork... [e00a6e0](https://github.com/astrolicious/studiocms/pull/77/commits/e00a6e09a849f0847f952c010e4c95b52012e73b)